management-proxy/session-manager/routes/system.js

```diff
@@ -58,8 +58,8 @@ module.exports = function (router) {
           if (response.statusCode == HttpStatus.OK && result.namespace) {
             var get_image_id_for_key = result.keys.reduce(function (obj, key) {
               // only compare images belonging to the current channel
-              // this takes care of images like ibuildthecloud/..., too
-              if (key.indexOf(channel) >= 0) {
+              // exclude buildstep as it's currently only available w/ tag :latest.
+              if (key.match(channel + "$") == channel && key.indexOf('buildstep:') === -1) {
```

Comment:
`$` matches against line end which for tag names makes sure that the array contains at most one element. The half-blind equality operator `==` takes care of the rest.

It's of course ugly. Something like `key.endsWith(channel)` would be much nicer – but that's ECMAScript 6 and I didn't feel like polyfilling this single case.